src/Microsoft.Framework.Configuration.Abstractions/IConfigurationRoot.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Configuration
+{
+    public interface IConfigurationRoot : IConfiguration
```

Comment:
Yes, it is the main motivation and we had it originally in all nodes but the behavior wasn't clear on child nodes: should it only affect the subtree starting here or the whole tree?